From fd7065730258aea3dbd16e357858639bc5e93c5f Mon Sep 17 00:00:00 2001
From: Jonas Gorski <jonas.gorski@bisdn.de>
Date: Wed, 16 Jun 2021 14:01:38 +0000
Subject: [PATCH] cmake use pkg-config for extracting pcre2 version

pcre2-config on yocto only outputs an error message to use pkg-config
instead, so let's do that.

Fixes the following build-error:

| ERROR: /usr/bin/pcre2-config should not be used, use an alternative such as pkg-config
| CMake Error at /tmp/jonas/work/corei7-64-poky-linux/libyang2/2.0.7-r0/recipe-sysroot-native/usr/share/cmake-3.16/Modules/FindPackageHandleStandardArgs.cmake:146 (message):
|   Could NOT find PCRE2: Found unsuitable version
|   "--should-not-have-used-/usr/bin/pcre2-config", but required is at least
|   "10.21" (found
|   /tmp/jonas/work/corei7-64-poky-linux/libyang2/2.0.7-r0/recipe-sysroot/usr/lib/libpcre2-8.so)
| Call Stack (most recent call first):
|   /tmp/jonas/work/corei7-64-poky-linux/libyang2/2.0.7-r0/recipe-sysroot-native/usr/share/cmake-3.16/Modules/FindPackageHandleStandardArgs.cmake:391 (_FPHSA_FAILURE_MESSAGE)
|   CMakeModules/FindPCRE2.cmake:19 (FIND_PACKAGE_HANDLE_STANDARD_ARGS)
|   CMakeLists.txt:307 (find_package)

Signed-off-by: Jonas Gorski <jonas.gorski@bisdn.de>
---
 CMakeModules/FindPCRE2.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeModules/FindPCRE2.cmake b/CMakeModules/FindPCRE2.cmake
index 8f60daa2..2f7c78d4 100644
--- a/CMakeModules/FindPCRE2.cmake
+++ b/CMakeModules/FindPCRE2.cmake
@@ -12,7 +12,7 @@ FIND_PATH(PCRE2_INCLUDE_DIR pcre2.h)
 FIND_LIBRARY(PCRE2_LIBRARY NAMES libpcre2.a pcre2-8)
 
 # Check required version
-execute_process(COMMAND pcre2-config --version OUTPUT_VARIABLE PCRE2_VERSION OUTPUT_STRIP_TRAILING_WHITESPACE)
+execute_process(COMMAND pkg-config --modversion libpcre2-8 OUTPUT_VARIABLE PCRE2_VERSION OUTPUT_STRIP_TRAILING_WHITESPACE)
 
 # Handle the QUIETLY and REQUIRED arguments and set PCRE_FOUND to TRUE if all listed variables are TRUE.
 INCLUDE(FindPackageHandleStandardArgs)
-- 
2.17.1

